flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -1203,6 +1221,16 @@ protected int getNumberRequiredTaskManagerSlots() {
 	//  Helper methods
 	// ------------------------------------------------------------------------
 
+	@VisibleForTesting
+	Configuration getFlinkConfig() {
+		return flinkConfig;
+	}
+
+	@VisibleForTesting
+	Collection<ResourceProfile> getSlotsPerWorker() {
+		return slotsPerWorker;
+	}
```

Comment:
Getting access to internal state is usually an indicator that our class under test is too powerful and is not well suited for testing.